File: WebApplication/src/modules/report/models/sheet-year.dto.ts
import { NomenclatureDto } from "src/infrastructure/models/nomenclature.dto";
import { SheetRowDto } from "./sheet-row.dto";
import { MonthDataDto } from "./month-data.dto";

export class SheetYearDto extends SheetRowDto {
  id: number;
  year: NomenclatureDto;

  sheetList: MonthDataDto[] = [];

  isExpand: boolean;
}

File: StudentCredit.Application/Reports/Dtos/SummaryReports/SheetYearDto.cs
using StudentCredit.Infrastructure.Helpers.Dtos;

namespace StudentCredit.Application.Reports.Dtos.SummaryReports
{
    public class SheetYearDto : SheetRowDto
    {
        public int Id { get; set; }

        public NomenclatureDto Year { get; set; }

        public List<MonthDataDto> SheetList { get; set; } = new List<MonthDataDto>();
    }
}
